File: src/m/solvers/cgsgamgoptions.py
from pairoptions import pairoptions
from collections import OrderedDict


def cgsgamgoptions(*args):

    #retrieve options provided in *args
    options = pairoptions(*args)
    solverOptions = OrderedDict()
    solverOptions['toolkit'] = 'petsc'
    solverOptions['mat_type'] = options.getfieldvalue('mat_type', 'mpiaij')
    solverOptions['ksp_type'] = options.getfieldvalue('ksp_type', 'cgs')
    solverOptions['pc_type'] = options.getfieldvalue('pc_type', 'gamg')

    return solverOptions

File: src/m/solvers/cgsgamgoptions.m
function solverOptions=cgsgamgoptions(varargin)

%CGSGAMGOPTIONS - define PETSc solver options for the Conjugate Gradient Squared (CGS) Krylov method with Geometric-Algebraic Multigrid (GAMG) preconditioning
%
%   Usage:
%      solverOptions=cgsgamgoptions(varargin);

%retrieve options provided in varargin
options=pairoptions(varargin{:});
solverOptions=struct();
solverOptions.toolkit='petsc';
solverOptions.mat_type=getfieldvalue(options, 'mat_type','mpiaij');
solverOptions.ksp_type=getfieldvalue(options, 'ksp_type','cgs');
solverOptions.pc_type=getfieldvalue(options, 'pc_type',  'gamg');

